invoiceregistry_test.go

```diff
@@ -464,18 +417,21 @@ func TestOverpayment(t *testing.T) {
 	first := c.testAmt / 2
 	rest := c.testAmt - first
 
-	notifyHtlc(2, first)
-	notifyHtlc(3, rest+1000)
+	c.notifyHtlc(rHash, invoice.PaymentAddr, optHtlcID(2),
+		optAmtPaid(first))
+
+	c.notifyHtlc(rHash, invoice.PaymentAddr, optHtlcID(3),
+		optAmtPaid(rest+1000))
 
-	c.checkHtlcFailed(<-resolved, ResultHtlcSetOverpayment)
+	c.checkHtlcFailed(ResultHtlcSetOverpayment)
 
 	// We should not have a second resolution, an update, or an accept here.
 	select {
 
 	case <-acceptChan:
 		require.FailNow(t, "Invoice accepted incorrectly")
 
-	case <-resolved:
+	case <-c.resolved:
```

Comment:
Side remark, not necessarily to be solved in this PR, but this check for unexpected resolutions could be made a standard part of `c.close()`, so that it is checked everywhere.

Reply:
I made the change in this PR.

Reply:
More clean up, good